source/StepBro.Core/Parser/FileBuilder.cs

```diff
@@ -875,6 +875,17 @@ public static int ParseFiles(ServiceManager services, ILogger logger, IScriptFil
                                 walker.Walk(listener, context);
                                 file.UpdateRootIdentifiers();
                             }
+                            catch (NotImplementedException e)
```

Comment:
This could easily catch any kind of exception, and add the info to a parsing error instead. Then the rest of the files would still be parsed, and this method can return in a nice way.